diff --git a/build-scripts/runtime_lib.cmake b/build-scripts/runtime_lib.cmake
index bac81c4..5fa77b0 100644
--- a/build-scripts/runtime_lib.cmake
+++ b/build-scripts/runtime_lib.cmake
@@ -187,7 +187,7 @@ file (GLOB header
 )
 LIST (APPEND RUNTIME_LIB_HEADER_LIST ${header})

-if (WAMR_BUILD_PLATFORM STREQUAL "windows")
+if (WAMR_BUILD_PLATFORM STREQUAL "windows" AND NOT MINGW)
     enable_language (ASM_MASM)
 else()
     enable_language (ASM)
